calico_containers/pycalico/datastore.py

```diff
@@ -134,7 +134,12 @@ class DatastoreClient(object):
 
     def __init__(self):
         etcd_authority = os.getenv(ETCD_AUTHORITY_ENV, ETCD_AUTHORITY_DEFAULT)
-        (host, port) = etcd_authority.split(":", 1)
+        if not validate_hostname_port(etcd_authority):
+            raise DataStoreError("Invalid %s. It must take the form "
+                                 "<address>:<port>. Value provided is '%s'" %
+                                 (ETCD_AUTHORITY_ENV, etcd_authority))
+        else:
```

Comment:
You don't need the `else` here - happy to let it go for now but is cleaner to not include it.